src/App.tsx

```diff
@@ -1,223 +1,102 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { AuthContext, AuthProvider } from './components/Auth/AuthContext';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+import { Filter } from './types/Filter';
+
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodosList/TodosList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/Error/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [title, setTitle] = useState('');
+  const [typeOfError, setTypeOfError] = useState(ErrorType.success);
+  const [filterType, setFilterType] = useState(Filter.all);
+
+  const handleGetRequest = (todoList: Todo[]) => {
+    setTodos(todoList);
+    setTypeOfError(ErrorType.success);
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(list => handleGetRequest(list))
+        .catch(() => setTypeOfError(ErrorType.get));
+    }
   }, []);
 
-  return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
+  useEffect(() => {
+    if (typeOfError !== ErrorType.success) {
+      setTimeout(() => setTypeOfError(ErrorType.success), 3000);
```

Comment:
You should clear the timeout on the component unmount. You can read here about it https://bobbyhadz.com/blog/react-cleartimeout